### PR TITLE
Feat/checkbox channeling

### PR DESCRIPTION
**Pull request by Stx69**
in ClientContext -> modified main getgotchis owned and getborrowed functions to merge lastChanneling value from
TheGraphApi.getGotchisGotchiverseInfoByIds
 
  add filter logic for checkbox in filters  "lastChanneled" in client owned and borrowed "lastChanneled" per gotchi <= today UTC 0h ->
  
image

### CODE REVIEW

File: src/pages/Client/store/effects/borrowed-gotchis.effect.ts [hunk 1]
```diff
@@ -18,10 +18,20 @@ export const onLoadBorrowedGotchis =
 
     TheGraphApi.getBorrowedByAddress(address)
       .then((borrowedGotchis: GotchiLending[]) => {
-        const sortedBorrowedGotchis: GotchiLending[] = CommonUtils.basicSort(borrowedGotchis, type, dir);
+        const promises: Promise<CustomAny>[] = borrowedGotchis.map((gotchi) => TheGraphApi.getGotchisGotchiverseInfoByIds([gotchi.id]));
```
Comment (skv2rock) [inline src/pages/Client/store/effects/borrowed-gotchis.effect.ts:21]: please change "CustomAny" to more specific type,  add type for gotchi

Comment (skv2rock) [reply src/pages/Client/store/effects/borrowed-gotchis.effect.ts:21]: also you calling "getGotchisGotchiverseInfoByIds" for each gotchi id, better would be create array with all ids and call it once

File: src/pages/Client/store/effects/borrowed-gotchis.effect.ts [hunk 2]
```diff
@@ -18,10 +18,20 @@ export const onLoadBorrowedGotchis =
...
+        Promise.all(promises).then((response) => {      
+          const modifiedBorrowed = borrowedGotchis.map(item => {
```
Comment (skv2rock) [inline src/pages/Client/store/effects/borrowed-gotchis.effect.ts:23]: please add types

File: src/pages/Client/store/effects/borrowed-gotchis.effect.ts [hunk 3]
```diff
@@ -18,10 +18,20 @@ export const onLoadBorrowedGotchis =
...
+          
+            const obj = response.find(o => o[0].id === item.id);
```
Comment (skv2rock) [inline src/pages/Client/store/effects/borrowed-gotchis.effect.ts:25]: please add types and rename const for mo specific name

Comment (skv2rock) [reply src/pages/Client/store/effects/borrowed-gotchis.effect.ts:25]: check other client effects

File: src/pages/Client/store/effects/borrowed-gotchis.effect.ts (hunk 1 above)
Comment (Stx69) [reply src/pages/Client/store/effects/borrowed-gotchis.effect.ts]: optimazed - created new GotchiLastChanneled share/model  specific Id as String 

new logic to speedup. Reading fetched ids as string and find LastChanneled per Gotchi.

File: src/pages/Client/store/effects/lent-gotchis.effect.ts [hunk 4]
```diff
@@ -16,16 +16,24 @@ export const onLoadLentGotchis =
...
+        lending.endTime = Number(lending.timeAgreed) + Number(lending.period);
+      });
+      const gotchiIds: string[] = lentGotchis.map((gotchi) => gotchi.id);
+
+      TheGraphApi.getGotchisGotchiverseInfoByIds(gotchiIds)
+        .then((gotchiIdsChanneled: GotchiLastChanneled[]) => {
+          const modifiedLent = lentGotchis.map((item) => {
+            const lastChanneled = gotchiIdsChanneled.find((o) => o.id === item.id);
+
+            return { ...item, ...lastChanneled };
```
Comment (skv2rock) [inline src/pages/Client/store/effects/lent-gotchis.effect.ts:30]: there no reason for destructuring objects, just add "lastChanneled" to item

File: src/pages/Client/store/effects/borrowed-gotchis.effect.ts [hunk 5]
```diff
@@ -16,12 +16,21 @@ export const onLoadBorrowedGotchis =
...
-        dispatch(borrowedGotchisSlices.loadBorrowedGotchisSucceded(sortedBorrowedGotchis));
-      })
-      .catch(() => dispatch(borrowedGotchisSlices.loadBorrowedGotchisFailed()))
-      .finally(() => dispatch(borrowedGotchisSlices.setIsInitialBorrowedGotchisLoading(false)));
+      TheGraphApi.getGotchisGotchiverseInfoByIds(gotchiIds)
+        .then((gotchiIdsChanneled: GotchiLastChanneled[]) => {
+          const modifiedBorrowed = borrowedGotchis.map((item) => {
+            const lastChanneled = gotchiIdsChanneled.find((o) => o.id === item.id);
+
+            return { ...item, ...lastChanneled };
```
Comment (skv2rock) [inline src/pages/Client/store/effects/borrowed-gotchis.effect.ts:27]: there no reason for destructuring objects, just add "lastChanneled" to item

**Comment by dudendy** [comment]
included in #589 and deployed to https://pilot.fireball.gg/

**Pull request closed without merging**